src/transformers/models/gptj/modeling_gptj.py

```diff
@@ -0,0 +1,892 @@
+        else:
+            raise ValueError(f"Input tensor rank should be one of [4, 5], but is: {len(tensor.shape)}")
+        new_shape = tensor.size()[:-2] + (num_attention_heads * attn_head_size,)
+        return tensor.view(new_shape)
+
+    def _attn(
+        self,
+        query,
+        key,
+        value,
+        causal_mask,
+        masked_bias,
+        attn_pdrop,
+        attention_mask=None,
+        head_mask=None,
+        scale_attn=None,
+    ):
+        # Keep the attention weights computation in fp32 to avoid overflow issues
+        query = query.to(torch.float32)
+        key = key.to(torch.float32)
+
+        attn_weights = torch.matmul(query, key.transpose(-1, -2))
+        attn_weights = torch.where(causal_mask, attn_weights, masked_bias.to(attn_weights.dtype))
+
+        if scale_attn is not None:
+            attn_weights = attn_weights / scale_attn
+
+        if attention_mask is not None:
+            # Apply the attention mask
+            attn_weights = attn_weights + attention_mask
+
+        attn_weights = nn.Softmax(dim=-1)(attn_weights)
+        attn_weights = attn_weights.to(value.dtype)
+        attn_weights = attn_pdrop(attn_weights)
+
+        # Mask heads if we want to
+        if head_mask is not None:
+            attn_weights = attn_weights * head_mask
+
+        attn_output = torch.matmul(attn_weights, value)
+
+        return attn_output, attn_weights
+
+
+class GPTJSelfAttention(nn.Module, GPTJAttentionMixin):
+    def __init__(self, config):
+        super().__init__()
+
+        self.window_size = None
+        max_positions = config.n_positions
+        bias = (
+            torch.tril(torch.ones((max_positions, max_positions), dtype=torch.uint8))
+            .view(1, 1, max_positions, max_positions)
+            .bool()
+        )
+
+        self.register_buffer(
+            "bias",
+            bias,
+        )
+
+        self.register_buffer("masked_bias", torch.tensor(-1e9))
+
+        self.attn_pdrop = nn.Dropout(config.attn_pdrop)
+        self.resid_pdrop = nn.Dropout(config.resid_pdrop)
+
+        self.embed_dim = config.n_embd
+        self.num_attention_heads = config.num_attention_heads
+        self.head_dim = self.embed_dim // self.num_attention_heads
+        self.register_buffer("scale_attn", torch.sqrt(torch.tensor(self.head_dim)))
+        if self.head_dim * self.num_attention_heads != self.embed_dim:
+            raise ValueError(
+                f"embed_dim must be divisible by num_attention_heads (got `embed_dim`: {self.embed_dim} and `num_attention_heads`: {self.num_attention_heads})."
+            )
+
+        self.k_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=False)
+        self.v_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=False)
+        self.q_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=False)
+        self.out_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=False)
+        self.rotary = config.rotary
```

Comment:
Note that the config does not have a `rotary` attribute (run `pytest tests/test_modeling_gptj.py` to check the tests locally).